Math/NumberTheory/Primes/Sieve/SmoothSieve.hs

```diff
@@ -0,0 +1,80 @@
+module Math.NumberTheory.Primes.Sieve.SmoothSieve
+  ( smoothData,
+    smoothNumbers,
+  ) where
+
+import Data.Vector (Vector, replicate, findIndices, accumulate, map)
```

Comment:
Usual pattern is to import unqualified only `Vector`, and use `V.replicate`, `V.findIndices`, `V.accumulate`, `V.map`).